1_fetch/src/get_daily_nwis_data.R

```diff
@@ -0,0 +1,63 @@
+get_daily_nwis_data <- function(site_info,parameter,stat_cd_select) {
+  #' 
+  #' @description Function to download NWIS daily data
+  #'
+  #' @param site_info data frame containing site info for NWIS daily site, including the variable "site_no"
+  #' @param parameter a character vector containing the USGS parameter codes of interest
+  #' @param stat_cd_select a character vector containing the USGS stat codes to retain
+  #'
+  #' @value A data frame containing daily values and data quality codes for each stat code (e.g. min/max/mean)
+  #' @examples 
+  #' get_daily_nwis_data(site_info = daily_sites,parameter="00300",stat_cd_select="00003")
+  
+  message(sprintf('Retrieving daily data for %s', site_info$site_no))
+
+  # Download daily data
+  site_data <- dataRetrieval::readNWISdv(
+    siteNumbers = site_info$site_no,parameterCd=parameter,statCd=stat_cd_select,startDate = "",endDate = "") %>%
+    dataRetrieval::renameNWISColumns(p00300="Value",p00095="Value")
+  
+  # Munge column names for some sites with different or unusually-named value columns
+  if(length(grep("Value_cd",names(site_data)))>1){
```

Comment:
If this gets to be very long, it could become a separate function that passes in the dataframe (or only the problem sites) and returns the processed data

Reply:
OK, yeah that probably makes sense. I will leave as is for now but can revisit if needed - perhaps I'm too optimistic that the number of 'unique' sites won't grow too much!